From c593ce79958bb95c3941d281b8e2ce51704e583d Mon Sep 17 00:00:00 2001
From: Tuukka Toivonen <tuukka.toivonen@intel.com>
Date: Wed, 17 Dec 2014 15:30:46 +0200
Subject: [PATCH 248/255] atomisp2: enable downscaling for capture pipe for
 ISP2401

Revise patch
"atomisp2: disable DS for capture pipe for some configuration"
so that downscaling is disabled only for older CSS versions
than ISP2401 new input system.

Change-Id: I78665e80d00c5220f1d71ff2bc900b8a3e8e4570
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-12780
Signed-off-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Signed-off-by: Tommi Franttila <tommi.franttila@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/36004
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 6d16c78..c401330 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2634,7 +2634,9 @@ static void __configure_capture_pp_input(struct atomisp_sub_device *asd,
 	ver_ds_factor = CEIL_DIV(height >> 1,
 			pipe_configs->output_info[0].res.height);
 
-	if (hor_ds_factor != ver_ds_factor) {
+	if (asd->isp->media_dev.hw_revision <
+	    (ATOMISP_HW_REVISION_ISP2401 << ATOMISP_HW_REVISION_SHIFT) &&
+	    hor_ds_factor != ver_ds_factor) {
 		dev_warn(asd->isp->dev,
 				"Cropping for capture due to FW limitation");
 		return;
-- 
1.9.1

